### ISSUE TITLE
Inconsistent identifier handling for ~Assets~ non-Concept Classes

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/concerto#241]
Identifier handling for Assets seems broken or at the minimum, confusing.

I would expect the two following types `A1` and `A2` to behave the same:
```
namespace org.test

asset A {
	o String name
}

asset A1 identified by id extends A {
	o String id
}

asset A2 identified by id {
	o String name
	o String id
}
```

But they don't:

**Validating A1**
```
{
    "$class": "org.test.A1",
    "name": "John"
}
```
yields:
```
{
  "$class": "org.test.A1",
  "id": "4b28e1e0-b4ad-4e3f-b311-e3ac77e93681",
  "name": "John",
  "$identifier": "4b28e1e0-b4ad-4e3f-b311-e3ac77e93681"
}
```

**Validating A2**
```
{
    "$class": "org.test.A2",
    "name": "John"
}
```
yields:
```
{
  "$class": "org.test.A2",
  "name": "John",
  "id": "5bd4f0b3-d394-4daa-8091-665018723fa7"
}
```

(where is the `$identifier` gone).

**Comment by jeromesimeon** [comment]
I could think of two different ways to handle this issue, both of which I think would be consistent:

1. treat all assets as having a built-in `identified` which means Validating A2 should validate with as:
   ```
   {
     "$class": "org.test.A2",
     "name": "John",
     "id": "5bd4f0b3-d394-4daa-8091-665018723fa7",
     "$identifier": "5bd4f0b3-d394-4daa-8091-665018723fa7"
   }
   ```
   This can be achieved with something like #240 
2. treat all assets as other classes, without `identified` 
   This can be achieved with something like #237

**Comment by jeromesimeon** [comment]
Same issue for: `Transaction` `Event` and `Participants`

I believe the problem is this code:
https://github.com/accordproject/concerto/blob/c2ecad3e5d52b0e8b1c7caf026d62abbf4f68b6f/packages/concerto-core/lib/introspect/identifieddeclaration.js#L51-L54

**Comment by jeromesimeon** [comment]
Addressed in #246:

```
zsh-5.8$ more test.cto
namespace org.test

asset A {
        o String name
}

asset A1 identified by id extends A {
        o String id
}

asset A2 identified by id {
        o String name
        o String id
}
zsh-5.8$ more test.json
{
    "$class": "org.test.A1",
    "id": "123",
    "name": "John"
}
zsh-5.8$ ./index.js validate --model test.cto --input test.json
2:19:40 PM - INFO: Input is valid
2:19:40 PM - INFO: 
{
  "$class": "org.test.A1",
  "id": "123",
  "name": "John",
  "$identifier": "123"
}
```

And:
```
zsh-5.8$ more test.json
{
    "$class": "org.test.A2",
    "id": "123",
    "name": "John"
}
zsh-5.8$ ./index.js validate --model test.cto --input test.json
2:20:24 PM - INFO: Input is valid
2:20:24 PM - INFO: 
{
  "$class": "org.test.A2",
  "name": "John",
  "id": "123",
  "$identifier": "123"
}
```
